File: InJava/app/src/main/java/com/barryzeha/simpleweather/mainModule/viewModel/MainViewModel.java
package com.barryzeha.simpleweather.mainModule.viewModel;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.barryzeha.simpleweather.common.entities.WeatherEntity;
import com.barryzeha.simpleweather.mainModule.model.MainRepository;

import javax.inject.Inject;

import dagger.hilt.android.lifecycle.HiltViewModel;

/****
 * Project Simple Weather
 * Created by Barry Zea H. on 21/01/2023
 * Copyright (c)  All rights reserved.
 ***/

@HiltViewModel
public class MainViewModel extends ViewModel {
    private final MainRepository repository;
    public @Inject
    MainViewModel (MainRepository repository){
        this.repository=repository;

    }

    public MutableLiveData<WeatherEntity> getResult() {return repository.getWeatherForecast();}
    public final   MutableLiveData<Integer> snackBarMsg = new MutableLiveData<>();
    public  MutableLiveData<Integer> getSnackBarMsg() {return snackBarMsg;}

    public final MutableLiveData<Boolean> loaded = new MutableLiveData<>(false);
    public  MutableLiveData<Boolean> getLoaded(){return loaded;}


    public void callWeatherForecast(double lat, double lon, String appId, String units, String lang){
             repository.callWeatherForecast(lat,lon,appId,units,lang);
    }


}

File: InKotlin/app/src/main/java/com/barryzeha/simpleweather/mainModule/viewModel/MainViewModel.kt
package com.barryzeha.simpleweather.mainModule.viewModel

import android.util.Log
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.barryzeha.simpleweather.R
import com.barryzeha.simpleweather.common.entities.WeatherEntity
import com.barryzeha.simpleweather.mainModule.model.MainRepository

import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.launch
import javax.inject.Inject

/****
 * Project Simple Weather
 * Created by Barry Zea H. on 18/01/2023
 * Copyright (c)  All rights reserved.
 ***/
@HiltViewModel
class MainViewModel @Inject constructor(private val repository: MainRepository):ViewModel()  {
    private val result = MutableLiveData<WeatherEntity>()
    fun getResult():LiveData<WeatherEntity> = result

    private val snackBarMsg = MutableLiveData<Int>()
    fun getSnackBarMsg() = snackBarMsg

    private val loaded = MutableLiveData<Boolean>()
    fun getLoaded() = loaded

    suspend fun getWeatherForecast(lat:Double, lon:Double, appId:String, units:String, lang:String){
        viewModelScope.launch {
            try {
                loaded.value= false
                val resultServer=repository.getWeatherForecast(lat,lon,appId, units, lang)
                result.value = resultServer
            } catch (e: Exception) {
                snackBarMsg.value = R.string.error_server_msg
                Log.e("ERROR", e.message.toString() )
            } finally {
                loaded.value = true
            }
        }
    }
}